batch/batch/front_end/front_end.py

```diff
@@ -888,122 +889,139 @@ async def create_jobs(request, userdata):
-                    '''
+                            job_parents_args,
+                        )
+                    except pymysql.err.IntegrityError as err:
+                        # 1062 ER_DUP_ENTRY https://dev.mysql.com/doc/refman/5.7/en/server-error-reference.html#error_er_dup_entry
+                        if err.args[0] == 1062:
+                            raise web.HTTPBadRequest(text=f'bunch contains job with duplicated parents ({err})')
+                        raise
+                    await tx.execute_many(
+                        '''
 INSERT INTO `job_attributes` (batch_id, job_id, `key`, `value`)
 VALUES (%s, %s, %s, %s);
 ''',
-                    job_attributes_args,
-                )
-
-                for inst_coll, resources in inst_coll_resources.items():
-                    n_jobs = resources['n_jobs']
-                    n_ready_jobs = resources['n_ready_jobs']
-                    ready_cores_mcpu = resources['ready_cores_mcpu']
-                    n_ready_cancellable_jobs = resources['n_ready_cancellable_jobs']
-                    ready_cancellable_cores_mcpu = resources['ready_cancellable_cores_mcpu']
+                        job_attributes_args,
+                    )
 
-                    await tx.execute_update(
+                    batches_inst_coll_staging_args = [
+                        (batch_id,
+                         inst_coll,
+                         rand_token,
+                         resources['n_jobs'],
+                         resources['n_ready_jobs'],
+                         resources['ready_cores_mcpu'])
+                        for inst_coll, resources in inst_coll_resources.items()]
+                    await tx.execute_many(
                         '''
 INSERT INTO batches_inst_coll_staging (batch_id, inst_coll, token, n_jobs, n_ready_jobs, ready_cores_mcpu)
 VALUES (%s, %s, %s, %s, %s, %s)
 ON DUPLICATE KEY UPDATE
-  n_jobs = n_jobs + %s,
-  n_ready_jobs = n_ready_jobs + %s,
-  ready_cores_mcpu = ready_cores_mcpu + %s;
+  n_jobs = n_jobs + VALUES(n_jobs),
+  n_ready_jobs = n_ready_jobs + VALUES(n_ready_jobs),
+  ready_cores_mcpu = ready_cores_mcpu + VALUES(ready_cores_mcpu);
 ''',
-                        (
-                            batch_id,
-                            inst_coll,
-                            rand_token,
-                            n_jobs,
-                            n_ready_jobs,
-                            ready_cores_mcpu,
-                            n_jobs,
-                            n_ready_jobs,
-                            ready_cores_mcpu,
-                        ),
-                    )
-                    await tx.execute_update(
+                        batches_inst_coll_staging_args)
+
+                    batch_inst_coll_cancellable_resources_args = [
+                        (batch_id,
+                         inst_coll,
+                         rand_token,
+                         resources['n_ready_cancellable_jobs'],
```

Comment:
I think you're missing a value here. I count 5 and there should be 6.

Reply:
The diff is a bit confusing. Each list is associated with the subsequent SQL, so this is associated with the `%s`'s on line 967.